### ISSUE TITLE
Expect .toBeResolved, .toBeRejected on promises

### ISSUE THREAD

**Issue by Nysosis** [alsatian-test/alsatian#384]
I'm doing some tests around methods which are returning promises, and it'd be nice to be able to check whether they're getting resolved/rejected in a nice succinct way (without potentially caring about the value of the resolved/rejected).

It could be achieved if #359 were to be completed with something like

```
export class MyClassTests {

	@Test()
	public MyMethod_Rejects_OnInvalidInput() {
		const sut = new MyClass()

		sut.MyMethod('abc').then(() => Expect.fail('Method should have failed'))
	}
}
```

However this seems a nicer way:

```
export class MyClassTests {

	@Test()
	public MyMethod_Rejects_OnInvalidInput() {
		const sut = new MyClass()

		Expect(sut.MyMethod('abc')).toBeRejected()
	}
}
```

I realise that the tests might need to become `AsyncTest`, and have `toBeRejected/toBeAccepted` return promises.

**Comment by jamesrichford** [comment]
Hey @Nysosis,

We've recently done done work on `toThorwAsync` which should fit your requirement I think.

```typescript
await Expect(async() => await sut.myMethod("abc")). toThrowAsync();
```

There's a little bit of extra syntax than your proposed solution because we need to not create the promise immediately and the check itself by nature is asynchronous.

This is currently in master but not published but I'm planning on shifting the karma release to unblock some smaller pieces that will be more use to people immediately :)

**Comment by jamesrichford** [comment]
@Nysosis this should be totally available in Alsatian 2.0.0 does this fit your requirement? :)

**Comment by jamesrichford** [comment]
Closing this as a duplicate, if you need something different than `toThrowAsync` please reopen or create a new issue @Nysosis :)
